pyartcd/pyartcd/jenkins.py

```diff
@@ -1,71 +1,107 @@
-import base64
 import logging
 import os
-import aiohttp
+from enum import Enum
+
+from jenkinsapi.jenkins import Jenkins
+
+from pyartcd import constants
 
 logger = logging.getLogger(__name__)
 
 
-async def trigger_jenkins_job(job_path: str, params: dict = {}):
+class Jobs(Enum):
+    BUILD_SYNC = 'aos-cd-builds/build%2Fbuild-sync'
+    MICROSHIFT = 'aos-cd-builds/build%2Fbuild-microshift'
+    OCP4 = 'aos-cd-builds/build%2Focp4'
+    RHCOS = 'aos-cd-builds/build%2Frhcos'
+
+
+logger.info('Initializing Jenkins client')
+jenkins = Jenkins(
+    constants.JENKINS_SERVER_URL,
+    username=os.environ['JENKINS_SERVICE_ACCOUNT'],
+    password=os.environ['JENKINS_SERVICE_ACCOUNT_TOKEN'],
+    lazy=True)
+logger.info('Connected to Jenkins %s', jenkins.version)
+
+
+def allow_backlinking(params: dict) -> dict:
     """
-    Trigger a job using remote API calls.
-    :param job_path: relative path to the job, starting from <buildvm hostname>:<jenkins port>
-    :param params: optional dict containing job parameters
+    If 'BUILD_URL' env var was provided, add a TRIGGERED_FROM
+    field to the build params to allow backlinking to the original job
     """
 
-    service_account = os.environ['JENKINS_SERVICE_ACCOUNT']
-    token = os.environ['JENKINS_SERVICE_ACCOUNT_TOKEN']
-
     build_url = os.environ.get('BUILD_URL', '')
     if build_url:
         params['TRIGGERED_FROM'] = build_url
+    return params
+
+
+def start_build_blocking(job_name: str, params: dict) -> str:
+    """
+    Starts a new build and blocks until it completes.
+
+    :param job_name: e.g. "build/build-sync", "scanning/scanning/images-health
+    :param params: a key-value collections to be passed to the build
+
+    Returns the build result
+    """
+
+    logger.info('Starting new build for job: %s', job_name)
+    job = jenkins.get_job(job_name)
+    build = job.invoke(build_params=params).block_until_building()
+    logger.info('Started new build for job %s',
+                build.baseurl.replace(constants.JENKINS_SERVER_URL, constants.JENKINS_UI_URL))
+
+    logger.info('Waiting for build to complete...')
+    build.block_until_complete()
+    logger.info(f'Completed with status: {build.get_status()}')
+    return build.get_status()
+
+
+def start_build_non_blocking(job_name: str, params: dict):
```

Comment:
Any strong reason to keep the 2 funcs separate vs `blocking: bool` param?

Reply:
Originally, I had a different implementation between the 2, so I had to keep them separate. I realize now they are pretty much the same, so makes sense to unify them.

Done in 085cc2e19d55d00fa9359c00c9eabde6d6de606b